src/Storage/Device/S3.php

```diff
@@ -524,7 +524,10 @@ private function call(string $method, string $uri, string $data = '')
         $response->code = \curl_getinfo($curl, CURLINFO_HTTP_CODE);
         
         if ($response->code >= 400) {
-            throw new Exception('HTTP request failed');
+            /* if (isset($response->headers['content-type']) && $response->headers['content-type'] == 'application/xml') {
```

Comment:
The error response seems to be XML but doesn't seem to decode properly.
@eldadfux ?

Reply:
Have no idea, you can just throw the body as a string - probably more safe.

Reply:
It is string.